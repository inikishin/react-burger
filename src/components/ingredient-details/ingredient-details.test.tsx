import React from "react";
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import IngredientDetails from "./ingredient-details";
import {Provider} from "react-redux";
import {BrowserRouter, Route} from "react-router-dom";

const mockStore = configureStore();


it('Тестируем ingredientDetails', () => {
    const initialState = {ingredients: {
        ingredients: [
            {_id: '1', name: 'Ingredient name', image_large: 'sime-img-url', calories: 100, proteins: 10, fat: 1, carbohydrates: 15}
        ]
    }};
    const store = mockStore(initialState);

    // TODO Вернуть после комментариев ментора
    // const tree = renderer
    //     .create(<Provider store={store}>
    //         <BrowserRouter>
    //             <Route match={{params: {id: '1'}}}>
    //                 <IngredientDetails/>
    //             </Route>
    //         </BrowserRouter>
    //     </Provider>)
    //     .toJSON();
    //
    // expect(tree).toMatchSnapshot();
});