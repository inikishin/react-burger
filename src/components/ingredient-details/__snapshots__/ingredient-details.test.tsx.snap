// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем ingredientDetails 1`] = `
<div
  className="bodyContent"
>
  <img
    alt="Ingredient name"
    src="sime-img-url"
  />
  <p
    className="text text_type_main-medium p-5"
  >
    Ingredient name
  </p>
  <p
    className="text text_type_main-default p-5"
  >
    Ingredient name
  </p>
  <div
    className="details"
  >
    <div
      className="secondaryText"
    >
      <p>
        Калории, ккал
      </p>
      <p
        className="text text_type_digits-default"
      >
        100
      </p>
    </div>
    <div
      className="secondaryText"
    >
      <p>
        Белки, г
      </p>
      <p
        className="text text_type_digits-default"
      >
        10
      </p>
    </div>
    <div
      className="secondaryText"
    >
      <p>
        Жиры, г
      </p>
      <p
        className="text text_type_digits-default"
      >
        1
      </p>
    </div>
    <div
      className="secondaryText"
    >
      <p>
        Углеводы, г
      </p>
      <p
        className="text text_type_digits-default"
      >
        15
      </p>
    </div>
  </div>
</div>
`;
